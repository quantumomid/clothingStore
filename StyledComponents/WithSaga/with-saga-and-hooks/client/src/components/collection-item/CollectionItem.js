import { useDispatch } from "react-redux";
import { addItem } from "../../redux/cart/cartActions";
import { AddCartButton, BackgroundImage, CollectionFooterContainer, CollectionItemContainer, NameContainer, PriceContainer } from "./collectionItemStyles";


function CollectionItem({ item }){
    const dispatch = useDispatch();
    const { name, price, imageUrl } = item
    return (
        <CollectionItemContainer>
            <BackgroundImage className='image' imageUrl={imageUrl} />
            <CollectionFooterContainer>
                <NameContainer>{name}</ NameContainer>
                <PriceContainer>{price}</ PriceContainer>
            </ CollectionFooterContainer>
            <AddCartButton onClick={() => dispatch(addItem(item))} inverted>Add to cart</AddCartButton>
        </ CollectionItemContainer>
    )
}

export default CollectionItem;