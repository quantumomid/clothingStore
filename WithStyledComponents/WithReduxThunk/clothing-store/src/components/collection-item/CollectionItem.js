import { connect } from "react-redux";
import { addItem } from "../../redux/cart/cartActions";
import { AddCartButton, BackgroundImage, CollectionFooterContainer, CollectionItemContainer, NameContainer, PriceContainer } from "./collectionItemStyles";


function CollectionItem({ item, addItem }){
    const { name, price, imageUrl } = item
    return (
        <CollectionItemContainer>
            <BackgroundImage className='image' imageUrl={imageUrl} />
            <CollectionFooterContainer>
                <NameContainer>{name}</ NameContainer>
                <PriceContainer>{price}</ PriceContainer>
            </ CollectionFooterContainer>
            <AddCartButton onClick={() => addItem(item)} inverted>Add to cart</AddCartButton>
        </ CollectionItemContainer>
    )
}

const mapDispatchToProps = dispatch => {
    return {
        addItem: item => dispatch(addItem(item))
    }
}

export default connect(null, mapDispatchToProps)(CollectionItem);