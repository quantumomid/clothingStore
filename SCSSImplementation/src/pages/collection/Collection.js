import "./Collection.scss";

import CollectionItem from "../../components/collection-item/CollectionItem";
import { connect } from "react-redux";
import { selectShopCollection } from "../../redux/shop/shopSelector";

const Collection = ({ collection }) => {
    // console.log(match.params.collectionId);
    // console.log(collection);
    const { title, items } = collection;
    return (
        <div className="collection-page">
            <h2 className="title">{ title }</h2>
            <div className="items">
                { items.map(item => <CollectionItem key={item.id} item={item} />) }
            </div>
        </div>
    )
}

// Second parameter here is the own props of the component i.e. Collection
// This is very handy as it will allow us access to "match" to use the collectionId!
const mapStateToProps = (state, ownProps) => ({
    collection: selectShopCollection(ownProps.match.params.collectionId)(state)
});

export default connect(mapStateToProps)(Collection);