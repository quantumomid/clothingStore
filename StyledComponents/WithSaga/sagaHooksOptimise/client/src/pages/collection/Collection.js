import CollectionItem from "../../components/collection-item/CollectionItem";
import { useSelector } from "react-redux";
import { selectShopCollection } from "../../redux/shop/shopSelector";
import { CollectionItemsContainer, CollectionPageContainer, CollectionTitle } from "./collectionStyles";
import { useParams } from "react-router-dom";

const Collection = () => {
    const { collectionId } = useParams();
    // console.log(collectionId);
    const { title, items } = useSelector(selectShopCollection(collectionId));    

    return (
        <CollectionPageContainer>
            <CollectionTitle>{ title }</CollectionTitle>
            <CollectionItemsContainer>
                { items.map(item => <CollectionItem key={item.id} item={item} />) }
            </CollectionItemsContainer>
        </CollectionPageContainer>
    )
}

export default Collection;