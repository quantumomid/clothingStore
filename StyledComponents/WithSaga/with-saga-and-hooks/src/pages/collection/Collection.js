import CollectionItem from "../../components/collection-item/CollectionItem";
import { connect, useSelector } from "react-redux";
import { selectShopCollection } from "../../redux/shop/shopSelector";
import { CollectionItemsContainer, CollectionPageContainer, CollectionTitle } from "./collectionStyles";

const Collection = ({ match }) => {
    // console.log(match.params.collectionId);
    const collection = useSelector(state => selectShopCollection(match.params.collectionId)(state));    
    const { title, items } = collection;
    return (
        <CollectionPageContainer>
            <CollectionTitle>{ title }</CollectionTitle>
            <CollectionItemsContainer>
                { items.map(item => <CollectionItem key={item.id} item={item} />) }
            </CollectionItemsContainer>
        </CollectionPageContainer>
    )
}

// // Second parameter here is the own props of the component i.e. Collection
// // This is very handy as it will allow us access to "match" to use the collectionId!
// const mapStateToProps = (state, ownProps) => ({
//     collection: selectShopCollection(ownProps.match.params.collectionId)(state)
// });

export default Collection;