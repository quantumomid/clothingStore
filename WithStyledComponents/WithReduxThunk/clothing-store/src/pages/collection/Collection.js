import CollectionItem from "../../components/collection-item/CollectionItem";
import { connect } from "react-redux";
import { selectShopCollection } from "../../redux/shop/shopSelector";
import { CollectionItemsContainer, CollectionPageContainer, CollectionTitle } from "./collectionStyles";

const Collection = ({ collection }) => {
    // console.log(match.params.collectionId);
    // console.log(collection);
    const { title, items } = collection;
    return (
        <CollectionPageContainer>
            <CollectionTitle>{ title }</CollectionTitle>
            <CollectionItemsContainer>
                { items.map(item => <CollectionItem key={item.id} item={item} />) }
            </CollectionItemsContainer>
        </CollectionPageContainer>
    )
}

// Second parameter here is the own props of the component i.e. Collection
// This is very handy as it will allow us access to "match" to use the collectionId!
const mapStateToProps = (state, ownProps) => ({
    collection: selectShopCollection(ownProps.match.params.collectionId)(state)
});

export default connect(mapStateToProps)(Collection);