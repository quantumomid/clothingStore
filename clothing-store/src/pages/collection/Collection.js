import "./Collection.scss";

import CollectionItem from "../../components/collection-item/CollectionItem";
import { connect } from "react-redux";
import { selectShopCollection } from "../../redux/shop/shopSelector";

const Collection = ({ collection }) => {
    // console.log(match.params.collectionId);
    console.log(collection);
    return (
        <div className="collection-page">
            <h2>COLLECTION PAGE</h2>
        </div>
    )
}

// Second parameter here is the own props of the component i.e. Collection
// This is very handy as it will allow us access to "match" to use the collectionId!
const mapStateToProps = (state, ownProps) => ({
    collection: selectShopCollection(ownProps.match.params.collectionId)(state)
});

export default connect(mapStateToProps)(Collection);