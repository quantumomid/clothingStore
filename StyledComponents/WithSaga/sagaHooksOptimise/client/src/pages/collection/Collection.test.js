import React from 'react';
import { mount } from 'enzyme';
import { Provider } from "react-redux";
import { mockStore } from '../../test/testUtils';
import Collection from "./Collection";
import { BrowserRouter } from "react-router-dom";
import Router from "react-router-dom";
import CollectionItem from "../../components/collection-item/CollectionItem";

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useParams: jest.fn(),
}));

const mockCollections = {
    hats: {
    id: 1,
    title: 'Hats',
    routeName: 'hats',
    items: [
      {
        id: 1,
        name: 'Brown Brim',
        imageUrl: 'https://i.ibb.co/ZYW3VTp/brown-brim.png',
        price: 25
      },
      {
        id: 2,
        name: 'Blue Beanie',
        imageUrl: 'https://i.ibb.co/ypkgK0X/blue-beanie.png',
        price: 18
      }
    ]
    },
    sneakers: {
        id: 2,
        title: 'Sneakers',
        routeName: 'sneakers',
        items: [
          {
            id: 10,
            name: 'Adidas NMD',
            imageUrl: 'https://i.ibb.co/0s3pdnc/adidas-nmd.png',
            price: 220
          },
          {
            id: 11,
            name: 'Adidas Yeezy',
            imageUrl: 'https://i.ibb.co/dJbG1cT/yeezy.png',
            price: 280
          },
          {
            id: 12,
            name: 'Black Converse',
            imageUrl: 'https://i.ibb.co/bPmVXyP/black-converse.png',
            price: 110
          }
        ]
      }
}

const store = mockStore({
    shop: {
        collections: mockCollections
    }
});



describe('Checkout Page Component', () => {
  let wrapper;

  beforeEach(() => {
    jest.spyOn(Router, 'useParams').mockReturnValue({ collectionId: 'hats' })
    wrapper = mount(
        <Provider store={store}>
            <BrowserRouter>
                <Collection />
            </BrowserRouter>
        </Provider>);
  });

    it('Should Render Checkout Page Component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the same number of CollectionItems as collection array', () => {
        expect(wrapper.find(CollectionItem).length).toBe(2);
    });
})