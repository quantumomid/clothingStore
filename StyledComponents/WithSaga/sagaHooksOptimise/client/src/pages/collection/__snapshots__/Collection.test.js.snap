// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout Page Component Should Render Checkout Page Component 1`] = `ReactWrapper {}`;
